app/javascript/components/pr-feed.vue

```diff
@@ -32,12 +35,47 @@
 </template>
 
 <script>
+
+import axios from 'axios';
+
 export default {
   props: {
     pullRequests: {
       type: Array,
       required: true,
     },
   },
+
+  data() {
+    return {
+      contador: 0,
+      prCopy: [...this.pullRequests],
```

Comment:
Quizás sería mejor tener un set donde guardes a cuáles PRs se les ha hecho like o un hash para guardar solo los totales en vez de copiar todos los PRs.

Reply:
@difernandez me gustaría saber qué pensai de esto, le comentaba a Miguel que quizás se podría hacer un set para guardar los ids de los PRs que se les ha hecho like en data y luego un computed para el total que considere los likes del prop + 1 si está en el set.

Pros: Ahorrarse tener que duplicar todos los PRs
Contras: Queda más enredado

Reply:
mm creo que yo suelo hacer una copia si tengo que agregarle un atributo o modificar uno de los existentes en los elementos de la lista. De hecho, no se si soy la mejor persona para pedir una segunda opinión, creo que esta parte la hicimos juntos con mike jaja.

El set no me convence, ya que habría que en el computed o method en que se calcule el total para un pr habría que hacer un `find` sobre ese set y se pierde la rapidez de acceso de un objeto.

Un punto medio que podría ser es lo otro que mencionas, un hash en la data que sea tipo `id: total`, algo así. Así necesitarías solo el id del pr para acceder al total sin hacer `find`. Ese me tinca más, pero creo que aún me inclino por la copia de la lista, me gusta tener centralizado en un objeto o array toda la info de las entidades.

Pero cómo dije esa parte la habíamos visto con mike si mal no recuerdo jaja así que si quieres otra opinión pueden preguntarle a alguien más

Reply:
> habría que hacer un find sobre ese set y se pierde la rapidez de acceso de un objeto.

El find en un set debiera ser O(1)

Otra pregunta, si cambia el prop de los PRs (digamos desde el padre) en este caso tampoco se actualizaría dentro de este componente cierto?

En todo caso si no te parece mal duplicar los PRs entonces yo lo dejaría hasta acá y que @miguelramirezv  decida según lo que le tinque más. A lo más si decide duplicar podría quedar como una tarjeta en futurología como posible mejora de performance.

Reply:
Ok lo dejo para futurología y lo converso con mi PO/scrum master

Reply:
> El find .has en un set debiera ser O(1)

cacha, no sabía lo del set, está bueno jaja

> Otra pregunta, si cambia el prop de los PRs (digamos desde el padre) en este caso tampoco se actualizaría dentro de este componente cierto?

Buen punto. Claro, si cambia la prop no se recalcularía la data. En este caso la prop viene de Rails, así que no podrían cambiar igual. Pero si, si se espera que cambien y el componente necesita tener en cuenta esto, entonces yo creo que habría que hacer algo de lo que mencionas, hash/set + computed. Otra opción en ese caso (y considerando que si la prop cambia entonces el padre sería un componente vue y no una vista rails) es hacer [lo que mencionan acá](https://forum.vuejs.org/t/is-mutating-object-props-bad-practice/17448/3), manejar todo con eventos al padre y que el padre se preocupe de hacer los cambios, el hijo solo recibe el objeto cambiado como prop